src/Geometry/coordinates.jl

```diff
@@ -43,32 +52,75 @@ end
 
 @pointtype Cartesian12Point x1 x2
 @pointtype Cartesian123Point x1 x2 x3
+
 # TODO: deprecate these
 const Cartesian2DPoint = Cartesian12Point
-const Cartesian3DPoint = Cartesian123Point
+@deprecate Cartesian2DPoint Cartesian12Point
 
+const Cartesian3DPoint = Cartesian123Point
+@deprecate Cartesian3DPoint Cartesian123Point
 
-product_coordinates(xp::XPoint{FT}, zp::ZPoint{FT}) where {FT} =
-    XZPoint{FT}(xp.x, zp.z)
-product_coordinates(xp::XPoint{FT}, z::FT) where {FT} = XZPoint{FT}(xp.x, z)
+product_coordinates(xp::XPoint, yp::YPoint) = XYPoint(promote(xp.x, yp.y)...)
+product_coordinates(xp::XPoint, zp::ZPoint) = XZPoint(promote(xp.x, zp.z)...)
 
-product_coordinates(xyp::XYPoint{FT}, zp::ZPoint{FT}) where {FT} =
-    XYZPoint{FT}(xyp.x, xyp.y, zp.z)
-product_coordinates(xyp::XYPoint{FT}, z::FT) where {FT} =
-    XYZPoint{FT}(xyp.x, xyp.y, z)
+product_coordinates(xyp::XYPoint, zp::ZPoint) =
+    XYZPoint(promote(xyp.x, xyp.y, zp.z)...)
 
 # TODO: get rid of these and refactor to consistent point types
 ZPoint(pt::Cartesian3Point{FT}) where {FT} = ZPoint{FT}(pt.x3)
-XYPoint(pt::Cartesian2DPoint{FT}) where {FT} = XYPoint{FT}(pt.x1, pt.x2)
-XYZPoint(pt::Cartesian3DPoint{FT}) where {FT} =
+XYPoint(pt::Cartesian12Point{FT}) where {FT} = XYPoint{FT}(pt.x1, pt.x2)
+XYZPoint(pt::Cartesian123Point{FT}) where {FT} =
     XYZPoint{FT}(pt.x1, pt.x2, pt.x3)
 
 Cartesian3Point(pt::ZPoint{FT}) where {FT} = Cartesian3Point{FT}(pt.z)
-Cartesian2DPoint(pt::XYPoint{FT}) where {FT} = Cartesian2DPoint{FT}(pt.x, pt.y)
-Cartesian3DPoint(pt::XYZPoint{FT}) where {FT} =
-    Cartesian3DPoint{FT}(pt.x, pt.y, pt.z)
-
-components(p::AbstractPoint) = SVector(ntuple(i -> getfield(p, i), nfields(p)))
+Cartesian12Point(pt::XYPoint{FT}) where {FT} = Cartesian12Point{FT}(pt.x, pt.y)
+Cartesian123Point(pt::XYZPoint{FT}) where {FT} =
+    Cartesian123Point{FT}(pt.x, pt.y, pt.z)
+
+ncomponents(p::AbstractPoint) = nfields(p)
+components(p::AbstractPoint{FT}, i::Symbol) where {FT} = getfield(p, i)::FT
+components(p::AbstractPoint{FT}, i::Integer) where {FT} = getfield(p, i)::FT
```

Comment:
Should this be singular?
```suggestion
component(p::AbstractPoint{FT}, i::Symbol) where {FT} = getfield(p, i)::FT
component(p::AbstractPoint{FT}, i::Integer) where {FT} = getfield(p, i)::FT
```

Reply:
I was thinking coordinates is kind of like `Base.axes` and there are not singular versions for the later, I can change it though although the similarity of the names `comoponents` `component` might be a bit error prone.